tests/integrationv2/test_client_authentication.py

```diff
@@ -129,97 +160,86 @@ def test_client_auth_with_s2n_client_no_cert(managed_process, cipher, curve, pro
-    for results in client.get_results():
-        assert results.exception is None
-        assert results.exit_code == 0
-
-    # Openssl should indicate the procotol version in a successful connection.
+    # Openssl should tell us that a cert was requested but not received
     for results in server.get_results():
         assert results.exception is None
         assert results.exit_code == 0
-        assert random_bytes in results.stdout
-        if protocol is Protocols.TLS13:
-            message = bytes("SSL_accept:SSLv3/TLS read client certificate\nSSL_accept:SSLv3/TLS read finished".encode('utf-8'))
-        else:
-            if 'openssl-1.0.2' in OpenSSL.get_version():
-                message = bytes('SSL_accept:SSLv3 read client certificate A\nSSL_accept:SSLv3 read client key exchange A\nSSL_accept:SSLv3 read certificate verify A\nSSL_accept:SSLv3 read finished A'.encode('utf-8'))
-            else:
-                message = bytes("SSL_accept:SSLv3/TLS read client certificate\nSSL_accept:SSLv3/TLS read client key exchange\nSSL_accept:SSLv3/TLS read change cipher spec\nSSL_accept:SSLv3/TLS read finished".encode('utf-8'))
+        assert b'write certificate request' in results.stderr
+        assert b'read client certificate' not in results.stderr
+        assert b"peer did not return a certificate" in results.stderr
+        assert_openssl_handshake_complete(results, False)
 
-        assert message in results.stderr
+    for results in client.get_results():
+        assert results.exception is None
+         # TLS1.3 OpenSSL fails after the handshake, but pre-TLS1.3 fails during
+        if protocol is not Protocols.TLS13:
+            assert (results.exit_code != 0) 
+            assert b"Failed to negotiate: 'TLS alert received'" in results.stderr
+            assert_s2n_handshake_complete(results, protocol, provider, False)
 
 
 @pytest.mark.uncollect_if(func=invalid_test_parameters)
-@pytest.mark.parametrize("cipher", ALL_TEST_CIPHERS, ids=get_parameter_name)
-@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
-@pytest.mark.parametrize("protocol", PROTOCOLS, ids=get_parameter_name)
-@pytest.mark.parametrize("certificate", ALL_TEST_CERTS, ids=get_parameter_name)
 @pytest.mark.parametrize("provider", [OpenSSL], ids=get_parameter_name)
-def test_client_auth_with_s2n_client_with_cert(managed_process, cipher, curve, protocol, provider, certificate):
+@pytest.mark.parametrize("protocol", PROTOCOLS, ids=get_parameter_name)
+@pytest.mark.parametrize("cipher", ALL_TEST_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", CERTS_TO_TEST, ids=get_parameter_name)
+@pytest.mark.parametrize("client_certificate", CERTS_TO_TEST, ids=get_parameter_name)
+def test_client_auth_with_s2n_client_with_cert(managed_process, cipher, protocol, provider, certificate, client_certificate):
     port = next(available_ports)
 
+    if protocol < Protocols.TLS12 and client_certificate.algorithm == 'EC':
```

Comment:
s/`EC`/`ECDSA`

Reply:
I can't change it just here. It would need to be changed everywhere. See https://github.com/awslabs/s2n/pull/2454#discussion_r538685581